File: problems/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        inds = {}
        for index,element in enumerate(nums):
            inds[element] = index
        for key in range(len(nums)):
            diff = target - nums[key]
            if(inds.get(diff) and inds.get(diff) != key):
                return [key, inds[diff]]

        

File: problems/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> data;
        for(int i = 0; i<nums.size(); i++) {
            data[nums[i]] = i;
        }
        for(int i=0; i<nums.size(); i++) {
            int temp = target-nums[i];
            if(data.contains(temp) && data[temp] != i) {
                return {i , data[temp]};
            }
        }
        return {0,1};
    }
};
